src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -2611,6 +2611,72 @@ Task Synchronization API can be used for multiple dynamic synchronization patter
 
 A complete example of Producer / Consumer, _ServiceDeployerConsumer_ can be downloaded link:examples/ServiceDeployerConsumer.xml[here].
 
+[[_task_signal_api]]
+=== Signal API
+
+The *Signal API* is a high-level API that enables users to interact with running jobs via signals (we also call them actions).
+That is, it allows a job to expose specific actions (e.g., soft stop, scale up, scale down) that can triggered by users from the *Workflow Execution portal*.
+
+More specifically, the Signal API allows a workflow to:
+
+  . expose signals, i.e., display them in the Workflow Execution portal (e.g., display the `soft_stop` signal)
+
+  . wait to receive these signals, or check for their reception regularly, and
+
+  . execute the corresponding behavior (defined in the workflow) at the signal reception.
+
+The Signal API uses the synchronization API to provide a *persistent key-value channel* to store job signals.
+It uses a single channel (called *PA_SIGNALS_CHANNEL*) to store the signals of all jobs. The signal channel contains an entry per job.
+The key of each entry is the job id, whereas the value of the entry is the set of signals used by the considered job.
+Job signals are organized in a link:https://docs.oracle.com/javase/8/docs/api/java/util/Set.html[Java Set] structure,
+which means that a signal appears only once in the set, even if it is sent many times.
+
+As already mentioned, the Signal API is based on the synchronization API. However, these APIs are intended for different purposes.
+The Synchronization API is used to exchange information (e.g., messages) between tasks and between jobs, whereas the
+Signal API is useful to exchange signals between a job and users (via the Workflow Execution portal or the REST API), it cannot be used to send signals from a job to another.
+
+==== Usage
+*Task Signal API* is a *Java API* defined by the following link:../javadoc/org/ow2/proactive/scheduler/signal/SignalApi.html[API documentation].
+
+The Signal API is available inside ProActive Tasks through the `signalapi` <<_glossary_script_bindings,script binding>>.
+
+It is easy to use and offers a set of basic operations on signals, summarized as follows:
```

Comment:
use and offers -> use and offer
This sentence is not clear "It is easy to use and offers a set of basic operations on signals, summarized as follows:"

Reply:
Changed

Reply:
Changed